client/hotrod-client/src/main/java/org/infinispan/client/hotrod/RemoteCacheManager.java

```diff
@@ -347,17 +347,34 @@ public RemoteCacheManagerAdmin administration() {
       return new RemoteCacheManagerAdminImpl(operationsFactory);
    }
 
+   public TransportFactory getTransportFactory() {
+      return transportFactory;
+   }
+
+   public Codec getCodec() {
+      return codec;
+   }
+
+
    @Override
    public void close() throws IOException {
       stop();
    }
 
-   private static class RemoteCacheKey {
+   public ClientListenerNotifier getListenerNotifier() {
+      return listenerNotifier;
+   }
+
+   public ExecutorService getAsyncExecutorService() {
+      return asyncExecutorService;
+   }
+
+   public static class RemoteCacheKey {
```

Comment:
This is because I needed to use is in my other class when I separated the code (as fas as I remember). 
I want just to speak with the others which will be the best way to go here

Reply:
I need to review all this as you comment bellow, but internally when I create he cache I need a cache that is K Collection<V> and the createRemoteCache creates a K,V cache

Reply:
but again, i did this a long time ago and i juste duplicated some of the parts in purpose to see how to deal with everything after

Reply:
the hotrod client code is hard to enter at first

Reply:
I'm reviewing everything now